addons/proton_scatter/src/cache/scatter_cache.gd

```diff
@@ -0,0 +1,129 @@
+@tool
+extends Node
+
+# ProtonScatterCacheNode
+#
+# Saves the transforms created by ProtonScatter nodes in an external resource
+# and restore them when loading the scene.
+#
+# Use this node when you don't want to wait for scatter nodes to fully rebuild
+# at start.
+# You can also enable "Show output in tree" to get the same effect, but the
+# cache makes it much more VCS friendly, and doesn't clutter your scene tree.
+
+const DEFAULT_CACHE_FOLDER := "res://addons/proton_scatter/cache/"
+
+const ProtonScatter := preload("res://addons/proton_scatter/src/scatter.gd")
+const ProtonScatterTransformList := preload("../common/transform_list.gd")
+
+
+@export_file("*.res", "*.tres") var cache_file := "":
+	set(val):
+		cache_file = val
+		update_configuration_warnings()
+
+# The resource where transforms are actually stored
+var _local_cache: ProtonScatterCacheResource
+var _scene_root: Node
+var _scatter_nodes: Array[ProtonScatter]
+
+
+func _ready() -> void:
+	if not is_inside_tree():
+		return
+
+	_scene_root = _get_local_scene_root(self)
+
+	# By default, Set the cache path to a file next to the current scene.
+	if cache_file.is_empty():
+		var scene_path: String = _scene_root.get_scene_file_path()
+		var scene_name: String
+
+		# Set a random name if we can't find the current scene
+		if scene_path.is_empty():
+			scene_name = str(randi())
+		else:
+			scene_name = scene_path.get_file().get_base_dir()
+			scene_name += "_" + str(scene_path.hash()) # Prevents name collisions
+
+		cache_file = DEFAULT_CACHE_FOLDER.get_basename().path_join(scene_name + "_scatter_cache.tres")
```

Comment:
What do you think about defaulting to .res? Most people will take the default until they are very familiar with the tool.

What about defaulting the name to start with the current scene name so the files are identifiable? If we use this we'll likely want to ship the xform cache with the game, which means the name needs to be identifiable. We can set it up manually, but maybe the default can be improved.

Reply:
+ We can default to res to save space yes. 
+ Cache names already have the scene name though? They also have the hash but that's in case you have two scenes with the same name in different folders, you don't want their cache to override each other.